api/src/main/java/org/openmrs/util/DatabaseUpdater.java

```diff
@@ -105,119 +121,125 @@ public static void executeChangelog() throws DatabaseUpdateException, InputRequi
 	 * @throws InputRequiredException
 	 */
 	public static void executeChangelog(String changelog, Map<String, Object> userInput) throws DatabaseUpdateException,
-	        InputRequiredException {
-		
+		InputRequiredException {
+
 		log.debug("Executing changelog: " + changelog);
-		
-		executeChangelog(changelog, userInput, null);
+
+		executeChangelog(changelog, userInput);
 	}
-	
+
 	/**
 	 * Interface used for callbacks when updating the database. Implement this interface and pass it
-	 * to {@link DatabaseUpdater#executeChangelog(String, Map, ChangeSetExecutorCallback)}
+	 * to {@link DatabaseUpdater#executeChangelog(String, ChangeSetExecutorCallback)}
 	 */
 	public interface ChangeSetExecutorCallback {
-		
+
 		/**
 		 * This method is called after each changeset is executed.
 		 *
 		 * @param changeSet the liquibase changeset that was just run
 		 * @param numChangeSetsToRun the total number of changesets in the current file
 		 */
-		public void executing(ChangeSet changeSet, int numChangeSetsToRun);
+		public void executing( ChangeSet changeSet, int numChangeSetsToRun );
 	}
-	
+
 	/**
-	 * Executes the given changelog file. This file is assumed to be on the classpath. If no file is
-	 * given, the default {@link #CHANGE_LOG_FILE} is ran.
+	 * Executes the given changelog file. This file is assumed to be on the classpath. 
 	 *
 	 * @param changelog The string filename of a liquibase changelog xml file to run
-	 * @param userInput nullable map from question to user answer. Used if a call to
-	 *            executeChangelog(&lt;String&gt;, null) threw an {@link InputRequiredException}
 	 * @return A list of messages or warnings generated by the executed changesets
 	 * @throws InputRequiredException if the changelog file requires some sort of user input. The
 	 *             error object will list of the user prompts and type of data for each prompt
 	 */
-	public static List<String> executeChangelog(String changelog, Map<String, Object> userInput,
-	        ChangeSetExecutorCallback callback) throws DatabaseUpdateException, InputRequiredException {
+	public static List<String> executeChangelog(String changelog, ChangeSetExecutorCallback callback)
+		throws DatabaseUpdateException, InputRequiredException {
 		log.debug("installing the tables into the database");
-		
-		if (changelog == null) {
-			changelog = CHANGE_LOG_FILE;
+
+		if ( changelog == null ) {
+			throw new IllegalArgumentException( "changelog must not be null" );
 		}
```

Comment:
The previous version of this class handles the null case by defaulting to a standard changelog. This is not longer possible with snapshots. Instead of silently decommissioning the handling of null and changing a contract of sorts callers are now informed that null is no longer an option.

Reply:
Ok, perfect!